From tbm@cyrius.com Sun Mar 19 17:06:33 2006
From: Martin Michlmayr <tbm@cyrius.com>
To: 355998@bugs.debian.org
Cc: jeroen@fox-toolkit.org, foxgui-users-request@lists.sourceforge.ne
Subject: GCC 4.1 compilation fix for fox version 1.4
X-Spam-Checker-Version: SpamAssassin 2.60-bugs.debian.org_2005_01_02 
	(1.212-2003-09-23-exp) on spohr.debian.org
X-Spam-Level: 
X-Spam-Status: No, hits=-5.0 required=4.0 tests=BAYES_00,VALID_BTS_CONTROL 
	autolearn=no version=2.60-bugs.debian.org_2005_01_02
X-CrossAssassin-Score: 2
MIME-Version: 1.0
Content-Transfer-Encoding: 8bit
Content-Type: text/plain; charset=iso-8859-1

tags 355998 + patch
thanks

The following patch makes fox1.4 compile with GCC 4.1.  Please see
http://bugs.debian.org/355998


--- ./include/FXString.h~	2006-03-20 00:34:59.000000000 +0000
+++ ./include/FXString.h	2006-03-20 00:35:06.000000000 +0000
@@ -26,6 +26,39 @@
 
 namespace FX {
 
+FXAPI FXint compare(const FXchar* s1,const FXchar* s2);
+FXAPI FXint compare(const FXchar* s1,const FXString& s2);
+FXAPI FXint compare(const FXString& s1,const FXchar* s2);
+FXAPI FXint compare(const FXString& s1,const FXString& s2);
+
+FXAPI FXint compare(const FXchar* s1,const FXchar* s2,FXint n);
+FXAPI FXint compare(const FXchar* s1,const FXString& s2,FXint n);
+FXAPI FXint compare(const FXString& s1,const FXchar* s2,FXint n);
+FXAPI FXint compare(const FXString& s1,const FXString& s2,FXint n);
+
+FXAPI FXint comparecase(const FXchar* s1,const FXchar* s2);
+FXAPI FXint comparecase(const FXchar* s1,const FXString& s2);
+FXAPI FXint comparecase(const FXString& s1,const FXchar* s2);
+FXAPI FXint comparecase(const FXString& s1,const FXString& s2);
+
+FXAPI FXint comparecase(const FXchar* s1,const FXchar* s2,FXint n);
+FXAPI FXint comparecase(const FXchar* s1,const FXString& s2,FXint n);
+FXAPI FXint comparecase(const FXString& s1,const FXchar* s2,FXint n);
+FXAPI FXint comparecase(const FXString& s1,const FXString& s2,FXint n);
+
+FXAPI FXString FXStringFormat(const FXchar* fmt,...) FX_PRINTF(1,2) ;
+FXAPI FXString FXStringVFormat(const FXchar* fmt,va_list args);
+
+FXAPI FXString FXStringVal(FXint num,FXint base);
+FXAPI FXString FXStringVal(FXuint num,FXint base);
+
+FXAPI FXString FXStringVal(FXlong num,FXint base);
+
+FXAPI FXString FXStringVal(FXfloat num,FXint prec,FXbool exp);
+FXAPI FXString FXStringVal(FXdouble num,FXint prec,FXbool exp);
+
+FXAPI FXint FXIntVal(const FXString& s,FXint base);
+FXAPI FXuint FXUIntVal(const FXString& s,FXint base);
 
 /**
 * FXString provides essential string manipulation capabilities.
--- ./include/FXQuatd.h~	2006-03-20 00:45:12.000000000 +0000
+++ ./include/FXQuatd.h	2006-03-20 00:46:40.000000000 +0000
@@ -30,6 +30,8 @@
 
 class FXMat3d;
 
+class FXQuatd;
+FXAPI FXQuatd arc(const FXVec3d& a,const FXVec3d& b);
 
 /// Double-precision quaternion
 class FXAPI FXQuatd : public FXVec4d {
--- ./include/FXQuatf.h~	2006-03-20 00:45:28.000000000 +0000
+++ ./include/FXQuatf.h	2006-03-20 00:46:29.000000000 +0000
@@ -30,6 +30,8 @@
 
 class FXMat3f;
 
+class FXQuatf;
+FXAPI FXQuatf arc(const FXVec3f& a,const FXVec3f& b);
 
 /// Single-precision quaternion
 class FXAPI FXQuatf : public FXVec4f {

-- 
Martin Michlmayr
http://www.cyrius.com/



